### PR TITLE
PHP error fixes

### PR DESCRIPTION
**Pull request by juggernautsei**
Below is a list of errors we were getting in our system and these are the changes that were made to clear up the error messages. 

[error_cleanup.txt](https://github.com/openemr/openemr/files/578080/error_cleanup.txt)

### CODE REVIEW

File: interface/globals.php [hunk 1]
```diff
@@ -98,7 +98,7 @@ function undoMagicQuotes($array, $topLevel=true) {
     $tmp = $_GET['site'];
   }
   else {
-    if (!$ignoreAuth) die("Site ID is missing from session data!");
+    if (!isset($ignoreAuth)) die("Site ID is missing from session data!");
```
Comment (bradymiller) [inline interface/globals.php]: I think using empty would be more accurate here(and should not throw a php error, although should test it):
``````
if (empty($ignoreAuth)) die("Site ID is missing from session data!");

Comment (bradymiller) [inline interface/globals.php]: Note I changed the !empty to empty above.

Comment (bradymiller) [inline interface/globals.php]: I also confirmed that no error will be thrown if the variable does not exist. So, should use empty here.

File: library/clinical_rules.php [hunk 2]
```diff
@@ -771,7 +771,7 @@ function test_rules_clinic($provider='',$type='',$dateTarget='',$mode='',$patien
     if ( (count($targetGroups) == 1) || ($mode == "report") ) {
 
       // If report itemization is turned on, then iterate the rule id iterator
-      if ($GLOBALS['report_itemizing_temp_flag_and_id']) {
+      if (isset($GLOBALS['report_itemizing_temp_flag_and_id'])) {
```
Comment (bradymiller) [inline library/clinical_rules.php]: use !empty
``````
if (!empty($GLOBALS['report_itemizing_temp_flag_and_id'])) {

File: library/clinical_rules.php [hunk 3]
```diff
@@ -792,7 +792,7 @@ function test_rules_clinic($provider='',$type='',$dateTarget='',$mode='',$patien
 
         $dateCounter = 1; // for reminder mode to keep track of which date checking
         // If report itemization is turned on, reset flag.
-        if ($GLOBALS['report_itemizing_temp_flag_and_id']) {
+        if (isset($GLOBALS['report_itemizing_temp_flag_and_id'])) {
```
Comment (bradymiller) [inline library/clinical_rules.php]: use !empty:
``````
if (!empty($GLOBALS['report_itemizing_temp_flag_and_id'])) {

File: library/clinical_rules.php [hunk 4]
```diff
@@ -877,7 +877,7 @@ function test_rules_clinic($provider='',$type='',$dateTarget='',$mode='',$patien
           $dateCounter++;
         }
         // If report itemization is turned on, then record the "failed" item if it did not pass
-        if ($GLOBALS['report_itemizing_temp_flag_and_id'] && !($temp_track_pass)) {
+        if (isset($GLOBALS['report_itemizing_temp_flag_and_id']) && !($temp_track_pass)) {
```
Comment (bradymiller) [inline library/clinical_rules.php]: use !empty
``````
if (!empty($GLOBALS['report_itemizing_temp_flag_and_id']) && !($temp_track_pass)) {

File: library/options.inc.php [hunk 5]
```diff
@@ -1717,7 +1717,7 @@ function generate_display_field($frow, $currvalue) {
   $data_type  = $frow['data_type'];
   $field_id   = isset($frow['field_id'])  ? $frow['field_id'] : null;
   $list_id    = $frow['list_id'];
-  $backup_list = $frow['list_backup_id'];
+  $backup_list = isset($frow['list_backup_id']) ? $frow['list_backup_id'] : null;
```
Comment (bradymiller) [inline library/options.inc.php]: this line looks good

File: library/classes/Controller.class.php [hunk 6]
```diff
@@ -92,7 +92,7 @@ function act($qarray) {
                $c_action = preg_replace("/[^A-Za-z0-9_]/","",array_pop($args));
                $args = array_reverse($args);
 
-               if(!call_user_func(array(Controller,"i_once"),$GLOBALS['fileroot'] ."/controllers/C_" . $c_name . ".class.php")) {
+               if(!call_user_func(array("Controller","i_once"),$GLOBALS['fileroot'] ."/controllers/C_" . $c_name . ".class.php")) {
```
Comment (bradymiller) [inline library/classes/Controller.class.php]: this line looks good

**Comment by bradymiller** [comment]
Review complete. Just some minor things to address. Let me know when it's ready for another review (should be ready to go into the codebase at that time).
thanks,
-brady

File: interface/globals.php (hunk 1 above)
Comment (juggernautsei) [inline interface/globals.php]: update has been commited

File: library/clinical_rules.php (hunk 2 above)
Comment (juggernautsei) [inline library/clinical_rules.php]: update has been commited

File: library/clinical_rules.php (hunk 4 above)
Comment (juggernautsei) [inline library/clinical_rules.php]: update has been commited

**Comment by juggernautsei** [comment]
Updates completed. There are a lot more of these to be done. Working through them as quickly as I can and doing the verification that the changes do no harm.

**Comment by bradymiller** [comment]
Just brought your fixes into the codebase. thanks for the contribution!
-brady

**Pull request merged**